templates/utils.js

```diff
@@ -2,45 +2,198 @@ import { inspect } from "util";
-const replaceByClonedSource = (options) => {
-  const clone = options.clone
-  return  (_target, source) => {
-    return clone(source)
-  }
-}
+const replaceByClonedSource = options => {
+  const clone = options.clone;
+  return (_target, source) => {
+    return clone(source);
+  };
+};
 
-export const deepMerge = createDeepMerge({ mergeArray: replaceByClonedSource });
+const deepMergeWithoutKeysOrder = createDeepMerge({ mergeArray: replaceByClonedSource });
+
+export const deepMerge = (...args) => {
+  const mergedConfig = deepMergeWithoutKeysOrder(...args);
+  const finalConfig = {};
+  for (const key of Object.keys(args[0])) {
+    finalConfig[key] = mergedConfig[key];
+  }
+  for (const key of Object.keys(mergedConfig)) {
+    if (!(key in finalConfig)) {
+      finalConfig[key] = mergedConfig[key];
+    }
+  }
+  return finalConfig;
+};
 
 export const withDefaults =
   (template, expectedArgsDefaults, debug = false) =>
-  (receivedArgs) => {
+  receivedArgs => {
     const argsWithDefault = Object.fromEntries(
-      Object.entries(expectedArgsDefaults)
-      .map(([argName, argDefault]) => [argName, receivedArgs[argName] ?? [argDefault]])
+      Object.entries(expectedArgsDefaults).map(([argName, argDefault]) => [
+        argName,
+        receivedArgs[argName] ?? [argDefault],
+      ]),
     );
 
     if (debug) {
       console.log(argsWithDefault, expectedArgsDefaults, receivedArgs);
     }
 
-    const expectedArgsNames = Object.keys(expectedArgsDefaults)
-    Object.keys(receivedArgs).forEach((receivedArgName) => {
+    const expectedArgsNames = Object.keys(expectedArgsDefaults);
+    Object.keys(receivedArgs).forEach(receivedArgName => {
       if (!expectedArgsNames.includes(receivedArgName)) {
         throw new Error(
-          `Templated received unexpected argument \`${receivedArgName}\`. Expecting only ${
-            expectedArgsNames.map(name => `\`${name}\``).join(", ")
-          }`
+          `Templated received unexpected argument \`${receivedArgName}\`. Expecting only ${expectedArgsNames
+            .map(name => `\`${name}\``)
+            .join(", ")}`,
         );
       }
     });
 
     return template(argsWithDefault);
   };
 
-export const stringify = val => {
-  const str = inspect(val, { depth: null, compact: true, maxArrayLength: null, maxStringLength: null });
+const addComments = (val, comments = {}) => {
+  let str = inspect(val, {
+    depth: null,
+    compact: false,
+    maxArrayLength: null,
+    maxStringLength: null,
+  });
+
+  if (Object.keys(comments).length === 0) {
+    return str;
+  }
+
+  // Add comments above their respective properties
+  // Create a map of all property paths in the object
+  const pathMap = new Map();
+
+  // Process flat properties first
+  Object.keys(comments)
+    .filter(key => !key.includes("."))
+    .forEach(prop => {
+      const regex = new RegExp(`(^|\\n)(\\s*)(${prop}):\\s`, "g");
```

Comment:
Nit: Maybe we also name the regexes nicely? 

```js
  // For top-level properties
  const findTopLevelPropertyPattern = new RegExp(`(^|\\n)(\\s*)(${prop}):\\s`, "g");
  // Matches: "\n  propertyName: "

  // For finding object start
  const findObjectStartPattern = new RegExp(`(^|\\n)(\\s*)(${currentPath}):\\s*\\{`, "g");
  // Matches: "\n  objectName: {"

  // For nested properties
  const findNestedPropertyPattern = new RegExp(`(\\n)(${expectedIndent})(${part}):\\s`, "g");
  // Matches: "\n    nestedProperty: "

```

Reply:
Yes, thanks! Updated, but named a bit different. `find` word looks redundant since regexes used mostly for finding, and changed `Pattern` to `Regex` to be more explicit [cc9a3a7](https://github.com/scaffold-eth/create-eth/pull/227/commits/cc9a3a734c10d004501b522fa24ee83d7e9db989)

Reply:
Make more sense 🙌